aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -754,6 +756,16 @@ void run_cudnn_SDP_bprop(
       "cuDNN SDPA expected grad_output.strides() == output.strides(), "
       "the previous step probably failed to materialize a grad_output "
       "with matching strides...");
+#else
+  const auto innermost_dO_stride = dO.strides()[dO.strides().size() - 1];
+  if (innermost_dO_stride != 1) {
+    TORCH_WARN_ONCE(
+        "cuDNN SDPA backward got grad_output with an innermost stride != 1 "
```

Comment:
I wonder if we should even warn once here since a lot of people still use this for testing and this is kind of just like log spams them and there's nothing more efficient they can do.